File: datastructures/arrays/python/algorithms/two_sum_2.py
def twoSum(numbers, target):
    i = 0
    j = len(numbers) - 1
    sum = 0

    while i < j:
        sum = numbers[i] + numbers[j]
        if sum > target:
            j -= 1
        elif sum < target:
            i += 1
        elif sum == target:
            return [i + 1, j + 1]

    return [-1, -1]


def main():
    numbers = [2, 7, 11, 15]
    target = 9

    print(twoSum(numbers, target))


if __name__ == "__main__":
    main()

File: datastructures/strings/java/algorithms/TwoSum2.java
import java.util.Arrays;

public class TwoSum2 {

    public static int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;
        int sum = 0;

        while (i < j) {
            sum = numbers[i] + numbers[j];
            if (sum > target) {
                j--;
            } else if (sum < target) {
                i++;
            } else {
                return new int[] { i + 1, j + 1 };
            }
        }

        return new int[] { -1, -1 };
    }

    public static void main(String[] args) {
        int[] numbers = { 2, 7, 11, 15 };

        int[] result = twoSum(numbers, 9);

        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + " ");
        }
    }
}
